runtime/src/main/java/org/corfudb/util/serializer/ProtobufSerializer.java

```diff
@@ -0,0 +1,132 @@
+package org.corfudb.util.serializer;
+
+import com.google.protobuf.Any;
+import com.google.protobuf.Message;
+
+import io.netty.buffer.ByteBuf;
+import io.netty.buffer.ByteBufInputStream;
+import io.netty.buffer.ByteBufOutputStream;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+
+import org.corfudb.runtime.CorfuRuntime;
+import org.corfudb.runtime.CorfuStoreMetadata.Record;
+import org.corfudb.runtime.collections.CorfuRecord;
+import lombok.extern.slf4j.Slf4j;
+
+@Slf4j
+public class ProtobufSerializer implements ISerializer {
```

Comment:
This is a pretty important component. Please put detailed comments.

Reply:
done.